setup.sh

```diff
@@ -50,6 +120,80 @@ if ! docker-compose version > /dev/null 2>&1; then
     fi
 fi
 
+# Check COMPOSE_PROFILES for existing configuration profile
+if get_profile "default"; then
+    PROFILE="default"
+elif get_profile "solar-only"; then
+    PROFILE="solar-only"
+else
+    # Prompt for configuration profile if not set
+    echo "Select configuration profile:"
+    echo ""
+    echo " 1 - default     (Powerwall w/ Gateway on LAN)"
+    echo " 2 - solar-only  (No Gateway - data retrieved from Tesla Cloud)"
+    echo ""
+    while :
+    do
+        read -r -p "Select profile: " response
+        if [ "${response}" == "1" ]; then
+            PROFILE="default"
+        elif [ "${response}" == "2" ]; then
+            PROFILE="solar-only"
+        else
+            continue
+        fi
+        # Add selected profile to COMPOSE_PROFILES
+        add_profile "${PROFILE}"
+        echo ""
+        break
+    done
+fi
+
+# Check if running as non-default user (not required for Windows Git Bash)
+if ! type winpty > /dev/null 2>&1; then
+    notset=0
+    if [ -z "${PWD_USER}" ]; then
+        PWD_USER="1000:1000"
+        notset=1
+    fi
+    if [ "${PWD_USER}" == "1000:1000" ]; then
+        desc="normal default"
+        name="Default"
+    else
+        desc="configured user"
+        name="  Saved"
+    fi
+    CURRENT_USER="$(id -u):$(id -g)"
+    if [ "${CURRENT_USER}" != "${PWD_USER}" ]; then
+        echo "WARNING: Your current user uid/gid does not match the ${desc}."
```

Comment:
Love this!

Reply:
I think this may help to solve a lot of potential issues, hopefully.

I tested on some different systems and ensured I was installing with a different uid/gid, and having this a prompt during setup to configure the system for that user solved those permission issues (or at the least will highlight the potential issues to the user).

Reply:
I would say that over half of the install related issues were related to this. This is gold! Thanks.